core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -142,28 +144,24 @@ object RequestChannel extends Logging {
       // the total time spent on authentication, which may be significant for SASL/SSL.
       recordNetworkThreadTimeCallback.foreach(record => record(networkThreadTimeNanos))
 
-      if (requestLogger.isDebugEnabled) {
+      if (isRequestLoggingEnabled) {
         val detailsEnabled = requestLogger.isTraceEnabled
-        def nanosToMs(nanos: Long) = TimeUnit.NANOSECONDS.toMicros(math.max(nanos, 0)).toDouble / TimeUnit.MILLISECONDS.toMicros(1)
-        val totalTimeMs = nanosToMs(endTimeNanos - startTimeNanos)
-        val requestQueueTimeMs = nanosToMs(requestDequeueTimeNanos - startTimeNanos)
-        val apiLocalTimeMs = nanosToMs(apiLocalCompleteTimeNanos - requestDequeueTimeNanos)
-        val apiRemoteTimeMs = nanosToMs(apiRemoteCompleteTimeNanos - apiLocalCompleteTimeNanos)
-        val apiThrottleTimeMs = nanosToMs(responseCompleteTimeNanos - apiRemoteCompleteTimeNanos)
-        val responseQueueTimeMs = nanosToMs(responseDequeueTimeNanos - responseCompleteTimeNanos)
-        val responseSendTimeMs = nanosToMs(endTimeNanos - responseDequeueTimeNanos)
-
-        requestLogger.debug(s"Completed request:${requestDesc(detailsEnabled)} from connection ${context.connectionId};" +
-          s"totalTime:$totalTimeMs," +
-          s"requestQueueTime:$requestQueueTimeMs," +
-          s"localTime:$apiLocalTimeMs," +
-          s"remoteTime:$apiRemoteTimeMs," +
-          s"throttleTime:$apiThrottleTimeMs," +
-          s"responseQueueTime:$responseQueueTimeMs," +
-          s"sendTime:$responseSendTimeMs," +
-          s"securityProtocol:${context.securityProtocol}," +
-          s"principal:${context.principal}," +
-          s"listener:${context.listenerName.value}")
+
+        val builder = new StringBuilder(256)
+        builder.append("Completed request:").append(requestDesc(detailsEnabled))
+          .append(",response:").append(response.responseAsString.getOrElse(""))
```

Comment:
Nice!

Reply:
Right although not sure how to avoid the `getOrElse`. We could fail instead of falling back the empty string though. Would that be better? Originally when I did this there were cases where it could be `None`, but that's no longer the case as you pointed out.

I thought we could do the detailed response thing separately since it's only useful if we change the response classes to take advantage of it. What do you think?

Reply:
I changed it to throw an exception since it should not happen.